content/src/tnips/tnip_2.md

```diff
@@ -0,0 +1,331 @@
+
+<p style="text-align:center;">C<sub>n+1</sub> = new committee from a deterministic shuffle</p>
+
+The final execution result is signed by the validator’s BLS12-381 private key and unreliably broadcast to all known validators ([CVVs](#committee-voting-validators-cvvs) and [NVVs](#non-voting-validators-nvvs)) through the gossip network.
+
+[NVVs](#non-voting-validators-nvvs) must also sign and vote on execution results at *e*.
+All validators continue to gossip signed execution results at *e* until the committee taking over for the next epoch (updated [C](#on-chain-committee-information)) reaches a quorum of signatures (2f+1) from all eligible validators.
+These votes comprise a quorum from all staked validator nodes on the network, completing the epoch.
+Pending [CVVs](#committee-voting-validators-cvvs) reliably forward all signed execution results to their fellow committee members to support a successful transition.
+
+##### Shuffle Mechanism
+
+The Fisher-Yates shuffle algorithm is used to randomly reorder validators. This shuffle occurs once per epoch, and uses the aggregate BLS signature from the leader certificate from the last committed consensus round mixed with the accumulated randomness during the epoch as the source of entropy.
+
+##### Leftover Certificates
+
+Transactions are executed if they’re included in the consensus DAG commit, so inevitably there will be transactions that were certified within one epoch but unsettled and executed before *e* is reached. The new committee should not rely on a previous committee’s certificates and must reach consensus again before executing any transactions leftover from a previous epoch. To ensure the best possible user experience, exiting [CVVs](#committee-voting-validators-cvvs) must track and reliably forward any remaining transactions to pending [CVVs](#committee-voting-validators-cvvs). [CVVs](#committee-voting-validators-cvvs) in the new epoch must reverify and prioritize these transactions in the early rounds of the new epoch.
+
+An alternative approach is for [CVVs](#committee-voting-validators-cvvs) to track their certified batches and continue advancing rounds until all their batches are executed. At *e<sub>t</sub>*, validators would stop proposing batches. Primaries would only propose empty headers to ensure additional commits to the DAG. Once all batches for a primary are settled, the primary would include a system message in their proposed headers to indicate the epoch should close. A quorum of these system messages triggers the epoch transaction.
+
+On the happy path, this seems more efficient. The network already spent computation and bandwidth to certify these transactions, and the additional overhead to propose empty blocks to reach consensus seems trivial compared to forwarding certified transactions to the next committee. The next committee would still need to re-verify all transactions, so the computation and bandwidth costs are duplicated for the next committee.
```

Comment:
If we allow for the primary's `max_delay_interval` * number_of_rounds * 2, we can achieve the same thing through time-based measurements

I think using time vs commited subdag rounds, we more reliably ensure epoch closures in case faulty leaders delay commits